aleph/logic/xref.py

```diff
@@ -272,12 +272,38 @@ def xref_collection(collection):
     )
     log.info(f"[{collection}] Clearing previous xref state....")
     delete_xref(collection, sync=True)
-    delete_entities(collection.id, origin=ORIGIN, sync=True)
+    delete_entities(collection.id, origin='xref', sync=True)
+
+    batch = []
+    batch_size = int(SETTINGS.XREF_SCROLL_SIZE)  # Ensure batch_size is an integer
+    for entity in iter_entities(collection_id=collection.id):
```

Comment:
Not super important and I’m not sure if this would have any measurable impact, but as you only need the ID and no other fields from ES, you could try using the `includes` argument with a value of `[]` (or maybe if that doesn’t work something like `["id"]` or `["_id"]`):

```suggestion
    for entity in iter_entities(collection_id=collection.id, includes=[]):
```